readme.md

```diff
@@ -276,9 +276,10 @@ limit.
 
 ## Server side connection upgrades
 
-*This section only applies to unsecure HTTP. In the case of HTTPS the protocol
-negotiation must always happen by the means of Application-Layer Protocol
-Negotiation (ALPN).*
+### Upgrading over HTTPS
+Over HTTPS, the server and client negotiate a protocol using Application Layer Protocol Negotiation. This is done before data is sent, so both the client and server know what protocol to use before handling the connection, so if we instruct the client to use HTTP/2, the connection is already HTTP/2 by the time we get it. Clever, isn't it!
```

Comment:
Please try to keep the style from the original markdown file: An empty line after the headerline, and 80chars per line.
I would also be in favor of removing the "Clever" part, since some readers might not be in favor of more personal remarks.

Reply:
Understandable, I only added this as I wanted to make my delivery seem less technical/robotic :P